nrpy/infrastructures/BHaH/seobnr/SEOBNR_waveform_C_codegen_library.py

```diff
@@ -0,0 +1,255 @@
+"""
+Set up C function library for SEOBNR inspiral integrations.
+
+Author: Zachariah B. Etienne
+        zachetie **at** gmail **dot* com
+"""
+
+from inspect import currentframe as cfr
+from types import FrameType as FT
+from typing import Union, cast
+
+import nrpy.c_codegen as ccg
+import nrpy.c_function as cfc
+import nrpy.equations.seobnr.BOB_aligned_spin_waveform_quantities as BOB
+import nrpy.helpers.parallel_codegen as pcg
+
+
+def register_CFunction_SEOBNRv5_NQC_corrections() -> Union[None, pcg.NRPyEnv_type]:
+    """
+    Register CFunction for evaluating the Non Quasi-Circular (NQC) corrections for the SEOBNRv5 waveform.
+
+    :return: None if in registration phase, else the updated NRPy environment.
+    """
+    if pcg.pcg_registration_phase():
+        pcg.register_func_call(f"{__name__}.{cast(FT, cfr()).f_code.co_name}", locals())
+        return None
+
+    includes = ["BHaH_defines.h", "BHaH_function_prototypes.h"]
+    desc = """Evaluate SEOBNRv5 NQC corrections."""
+    cfunc_type = "int"
+    name = "SEOBNRv5_aligned_spin_NQC_corrections"
+    params = "commondata_struct *restrict commondata"
+    bob_wf = BOB.BOB_aligned_spin_waveform_quantities()
+
+    body = """
+REAL times[commondata->nsteps_fine], Q1[commondata->nsteps_fine], Q2[commondata->nsteps_fine], Q3[commondata->nsteps_fine], P1[commondata->nsteps_fine], P2[commondata->nsteps_fine];
```

Comment:
Just how large is `nsteps_fine`? Declaring memory on the stack can be dangerous. Usually I'd only do this if I were certain `nsteps_fine` were no larger than 128 or so.

Reply:
Hi Zazh,

Thanks for pointing this out! 
I've replaced all large array declarations with dynamical allocation (calloc).

Cheers,
Sid